R/toolApplyScenPrefTrends.R

```diff
@@ -31,8 +31,14 @@ toolApplyScenPrefTrends <- function(baselinePrefTrends, scenParPrefTrends, GDPpc
   GDPpcMER[, regionCat := ifelse(region %in% individualReg, region, regionCat)]
   mitigationFactors <- merge(mitigationFactors, GDPpcMER, by = "regionCat", allow.cartesian = TRUE, all.x = TRUE)[, regionCat := NULL]
   # apply mitigation factors
+
+  checkMitigation <- copy(baselinePrefTrends)
+  setnames(checkMitigation, "value", "old")
   PrefTrends <- merge(baselinePrefTrends, mitigationFactors, by = c("region", "level", "subsectorL1", "subsectorL2", "vehicleType", "technology"), all.x = TRUE, allow.cartesian = TRUE)
   PrefTrends[period  >= policyStartYear & !is.na(target), value := value * applyLogisticTrend(period, target, symmyr, speed)][, c("target", "symmyr", "speed") := NULL]
+  check <- merge(checkMitigation, PrefTrends, by = intersect(names(checkMitigation), names(PrefTrends)), all = TRUE)
+  check[, diff := abs(value - old)]
+  if (max(check$diff) < 0.001) stop("Mitigation preference factors have not been applied correctly. Please check toolApplyScenPrefTrends()")
```

Comment:
does this need to be > instead of <?

Reply:
No it's supposed to check whether mitigation factors were applied. When the difference is very small (or zero) no mitigation factors were applied and the preftrends equal the baseline :) The whole function is only called, when the mitigationfactors are not NULL (so not if the scenario is Mix1)

Reply:
ok thanks!